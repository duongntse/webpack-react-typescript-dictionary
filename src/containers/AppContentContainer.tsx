export { default as AppContentContainer } from 'src/modules/content/AppContent';
