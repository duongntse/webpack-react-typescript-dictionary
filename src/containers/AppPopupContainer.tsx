import {
    connect
    // ConnectedProps
} from 'react-redux';
import AppPopup from 'src/modules/popup/AppPopup';
import { setSearchBoxAbility } from 'src/store/searchBox/actions';
import { DISABLE } from 'src/store/searchBox/constants';
import { IStoreState } from 'src/store/types';
const mapStateToProps = (state: IStoreState) => {
    return {
        disabled: DISABLE == state.searchBox.ability ? true : false,
    };
};
const mapDispatchToProps = (dispatch: any) => ({
    setSearchBox: (ability: 'ENABLE' | 'DISABLE') => {
        dispatch(setSearchBoxAbility(ability));
    }
});

const connector = connect(mapStateToProps, mapDispatchToProps);

// type PropsFromRedux = ConnectedProps<typeof connector>;

// type Props = PropsFromRedux;

export default connector(AppPopup);
