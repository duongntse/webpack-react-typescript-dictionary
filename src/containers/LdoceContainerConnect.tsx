import { connect } from 'react-redux';
import { setSearchBoxAbility } from 'src/store/searchBox/actions';
// import { LdoceContainer } from 'src/modules/ldoce/';
import LdoceContainer from 'src/modules/ldoce/component/container/LdoceContainer';
const mapStateToProps = (state: any, ownProps: any) => {
	return {
		...state,
		...ownProps,
	};
};
const mapDispatchToProps = (dispatch: any) => ({
	setSearchBox: (ability: 'ENABLE' | 'DISABLE') => {
		dispatch(setSearchBoxAbility(ability));
	},
});

// const connector = connect(mapStateToProps, mapDispatchToProps);

export { default as LdoceContainer } from 'src/modules/ldoce/component/container/LdoceContainer';
export default connect(mapStateToProps, mapDispatchToProps)(LdoceContainer);
// export default connector(LdoceContainer);
