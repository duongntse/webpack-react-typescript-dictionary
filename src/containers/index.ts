export { default as SpeakerContainer } from './SpeakerContainer';
export { default as AppPopupContainer } from './AppPopupContainer';
export { default as LdoceContainerConnect } from './LdoceContainerConnect';
export * from './AppContentContainer';
