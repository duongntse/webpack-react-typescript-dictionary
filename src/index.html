<!DOCTYPE html>
<html>
	<head>
		<meta charset="UTF-8" />
		<title>Hello React!</title>
		<link
			rel="stylesheet"
			href="https://fonts.googleapis.com/css?family=Roboto:300,400,500,700&display=swap"
		/>
	</head>
	<body>
		<noscript>You need to enable JavaScript to run this app.</noscript>
		<div id="root"></div>
		<div id="example"></div>

		<!-- Dependencies -->
		<!-- react, react-dom -->

		<!-- <script
			src="https://unpkg.com/react@16/umd/react.production.min.js"
			crossorigin
		></script>
		<script
			src="https://unpkg.com/react-dom@16/umd/react-dom.production.min.js"
			crossorigin
		></script> -->
		<!-- react, react-dom -->
		<!-- <script
			crossorigin
			src="https://unpkg.com/react@16/umd/react.development.js"
		></script>
		<script
			crossorigin
			src="https://unpkg.com/react-dom@16/umd/react-dom.development.js"
		></script> -->
		<!-- lodash -->
		<script src="https://cdnjs.cloudflare.com/ajax/libs/lodash.js/4.17.15/lodash.core.js"></script>
		<!-- Jquery -->
		<script
			src="https://code.jquery.com/jquery-3.1.0.js"
			integrity="sha256-slogkvB1K3VOkzAI8QITxV3VzpOnkeNVsKvtkYLMjfk="
			crossorigin="anonymous"
		></script>
		<!-- Main -->
		<!-- <script src="./bundle.js"></script> -->
	</body>
</html>
