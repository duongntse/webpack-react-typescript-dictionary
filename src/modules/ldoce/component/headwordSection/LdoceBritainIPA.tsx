import React from 'react';
import { Typography } from '@material-ui/core';
import { withStyles, createStyles, WithStyles } from '@material-ui/core/styles';

const styles = () =>
    createStyles({
        ipa: {
            fontSize: '1em'
        }
    });

type StyleProps = WithStyles<typeof styles>;
type Props = { br_ipa?: string } & StyleProps;

const LdoceBritainIPA = withStyles(styles)(
    class extends React.Component<Props, {}> {
        constructor(props: Props) {
            super(props);
        }

        render() {
            const { classes } = this.props;
            const { br_ipa } = this.props;
            return (
                <Typography
                    className={classes.ipa}
                    style={{ fontSize: '1em' }}
                    display='inline'
                    variant='body1'
                    component='p'
                    color='textSecondary'>
                    {br_ipa ? `/${br_ipa}/` : ''}
                </Typography>
            );
        }
    }
);

export default LdoceBritainIPA;
