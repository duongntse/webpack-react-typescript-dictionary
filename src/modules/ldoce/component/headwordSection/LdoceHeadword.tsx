import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import { Typography } from '@material-ui/core';

const useStyles = makeStyles({
    headword: {
        fontSize: '1.8em'
    }
});

type Props = { headword?: string };

export default function LdoceHeadword(props: Props) {
    const classes = useStyles();
    const { headword } = props;
    return (
        <Typography
            className={`textHeadword ${classes.headword}`}
            display='inline'
            variant='h4'
            component='h4'
            color='textSecondary'>
            {headword || ' '}
        </Typography>
    );
}
