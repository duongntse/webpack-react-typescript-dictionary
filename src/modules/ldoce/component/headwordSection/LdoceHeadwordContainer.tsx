// LdoceHeadwordSection
import React from 'react';
import { createStyles, WithStyles, withStyles } from '@material-ui/core/styles';
import { Theme } from '@material-ui/core/styles/createMuiTheme';
import { Grid } from '@material-ui/core';
import LdoceHeadword from './LdoceHeadword';
import LdoceBritainIPA from './LdoceBritainIPA';
// import { Speaker } from 'src/modules/common';
import SpeakerContainer from 'src/containers/SpeakerContainer';
import { SpeakerType } from 'src/modules/utils';
import { API_LDOCE_DOMAIN } from 'src/modules/utils';
import { BRITAIN_SPEAKER_HEADWORD } from 'src/modules/utils';

const styles = (theme: Theme) =>
    createStyles({
        notifyRoot: {
            display: 'flex',
            flexWrap: 'wrap',
            justifyContent: 'center',
            overflow: 'hidden',
            padding: 4
        },
        root: {
            backgroundColor: theme.palette.background.paper,
            padding: 4
        }
    });

type Props = {
    headword: string;
    br_ipa?: string;
    br_audio_url?: string;
} & WithStyles<typeof styles>;
type State = {};
const LdoceHeadwordContainer = withStyles(styles)(
    class extends React.Component<Props, State> {
        constructor(props: Props) {
            super(props);
            this.state = {};
        }

        render() {
            const { classes } = this.props;

            const { headword, br_ipa, br_audio_url } = this.props;
            const audioUrlAttachedDomain =
                (br_audio_url && `${API_LDOCE_DOMAIN}${br_audio_url}`) || '';

            return (
                <Grid className={classes.root} item xs={12}>
                    <LdoceHeadword headword={headword} />
                    {` `}
                    <LdoceBritainIPA br_ipa={br_ipa} />
                    {` `}
                    <SpeakerContainer
                        speakerName={BRITAIN_SPEAKER_HEADWORD}
                        autoPlay={true}
                        isIPA={true}
                        br_audio_url={audioUrlAttachedDomain}
                        speakerType={
                            (br_audio_url && SpeakerType.LDOCE) ||
                            SpeakerType.GOOGLE
                        }
                    />
                </Grid>
            );
        }
    }
);
export default LdoceHeadwordContainer;
// export default withStyles(styles)(LdoceHeadwordContainer);
