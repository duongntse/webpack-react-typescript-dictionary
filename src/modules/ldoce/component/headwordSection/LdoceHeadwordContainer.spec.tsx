import React from 'react'
// test file
import LdoceHeadwordContainer from './LdoceHeadwordContainer';
import LdoceHeadword from './LdoceHeadword';
import LdoceBritainIPA from './LdoceBritainIPA';
import { SpeakerContainer } from 'src/containers';

import ReactTestRenderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';

const mockStore = configureStore([]);

import {
    // shallow, 
    mount,
    // render, 
} from 'enzyme';



function setup() {
    const props = {
        headword: 'test', br_ipa: 'test', br_audio_url: 'test'
    }
    let store: any;
    store = mockStore({
        searchBox: { ability: 'ENABLE' }
    });
    const renderer = ReactTestRenderer.create(
        <Provider store={store}>
            <LdoceHeadwordContainer {...props} />
        </Provider>
    );

    const wrapper = mount(
        <Provider store={store}>
            <LdoceHeadwordContainer {...props} />
        </Provider>
    );

    return { props, store, renderer, wrapper }
}

describe('<LdoceHeadwordContainer />', () => {
    const { wrapper, renderer } = setup();

    it('should render with given state from redux store', () => {
        const tree = renderer.toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('should render the inner LdoceHeadword', () => {

        const foundedComponent = wrapper.find(LdoceHeadword);
        expect(foundedComponent.length).toEqual(1);
    });
    it('should render the inner LdoceBritainIPA', () => {

        const foundedComponent = wrapper.find(LdoceBritainIPA);
        expect(foundedComponent.length).toEqual(1);
    });

    it('should render the inner SpeakerContainer', () => {
        const foundedComponent = wrapper.find(SpeakerContainer);
        expect(foundedComponent.length).toEqual(1);
    });

})
