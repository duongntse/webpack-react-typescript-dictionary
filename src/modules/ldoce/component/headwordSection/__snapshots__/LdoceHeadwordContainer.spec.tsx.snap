// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LdoceHeadwordContainer /> should render with given state from redux store 1`] = `
<div
  className="MuiGrid-root Component-root-2 MuiGrid-item MuiGrid-grid-xs-12"
>
  <h4
    className="MuiTypography-root textHeadword makeStyles-headword-106 MuiTypography-h4 MuiTypography-colorTextSecondary MuiTypography-displayInline"
  >
    test
  </h4>
   
  <p
    className="MuiTypography-root Component-ipa-137 MuiTypography-body1 MuiTypography-colorTextSecondary MuiTypography-displayInline"
    style={
      Object {
        "fontSize": "1em",
      }
    }
  >
    /test/
  </p>
   
  <div
    className="MuiBox-root MuiBox-root-144 Component-boxMedia-138 boxAudioMedia BRITAIN_SPEAKER_HEADWORD"
    onClick={[Function]}
  >
    <button
      aria-label="delete"
      className="MuiButtonBase-root MuiIconButton-root Component-iconButton-139  MuiIconButton-colorPrimary"
      disabled={false}
      onBlur={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiIconButton-label"
      >
        <audio
          autoPlay={false}
          className="MuiCardMedia-root MuiCardMedia-media"
          preload="auto"
          src="https://api.pearson.comtest"
        />
        <svg
          aria-hidden="true"
          className="MuiSvgIcon-root Component-speakerIcon-140 BRITAIN_SPEAKER_HEADWORD"
          focusable="false"
          role="presentation"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm-2 14.5v-9l6 4.5-6 4.5z"
          />
        </svg>
      </span>
      <span
        className="MuiTouchRipple-root"
      />
    </button>
  </div>
</div>
`;
